eoflow/models/metrics.py

```diff
@@ -156,3 +164,144 @@ def get_config(self):
+        self.edge_func_params = edge_func_params
+        self.pixel_size = pixel_size
+
+    def update_state(self, reference: np.ndarray, measurement: np.ndarray, encode_reference: bool = True,
+                     background_value: int = 0) -> None:
+        """ Calculate the error metrics for a measurement and reference arrays. For each .
+
+        If encode_reference is set to True, connected components will be used to label objects in the reference and
+        measurements.
+        """
+
+        if not tf.executing_eagerly():
+            warnings.warn("Geometric metrics must be run with eager execution. If running as a compiled Keras model, "
+                          "enable eager execution with model.run_eagerly = True")
+
+        reference = reference.numpy() if isinstance(reference, tf.Tensor) else reference
+        measurement = measurement.numpy() if isinstance(reference, tf.Tensor) else measurement
+
+        self._validate_input(reference, measurement)
+
+        for ref, meas in zip(reference, measurement):
+            ref = ref
+            meas = meas
+
+            if encode_reference:
+                cc_reference = measure.label(ref, background=background_value)
+            else:
+                cc_reference = ref
+
+            cc_measurement = measure.label(meas, background=background_value)
+            components_reference = set(np.unique(cc_reference)).difference([background_value])
+
+            ref_edges = self.edge_func(cc_reference)
+            meas_edges = self.edge_func(cc_measurement)
+            for component in components_reference:
+                reference_mask = cc_reference == component
+
+                uniq, count = np.unique(cc_measurement[reference_mask & (cc_measurement != background_value)],
+                                        return_counts=True)
+                ref_area = np.sum(reference_mask)
+
+                max_interecting_measurement = uniq[count.argmax()] if len(count) > 0 else background_value
+                meas_mask = cc_measurement == max_interecting_measurement
+                meas_area = np.count_nonzero(cc_measurement == max_interecting_measurement)
+                intersection_area = count.max() if len(count) > 0 else 0
+
+                self.oversegmentation_error.append(self._segmentation_error(intersection_area, ref_area))
+                self.undersegmentation_error.append(self._segmentation_error(intersection_area, meas_area))
+                border_ref_edge = ref_edges.squeeze() & reference_mask.squeeze()
+                border_meas_edge = meas_edges.squeeze() & meas_mask.squeeze()
+
+                self.border_error.append(self._border_err(border_ref_edge, border_meas_edge))
+                self.fragmentation_error.append(self._fragmentation_err(len(uniq), reference_mask))
+
+    def get_oversegmentation_error(self) -> float:
+        """ Return oversegmentation error. """
+        return np.array(self.oversegmentation_error).mean()
```

Comment:
In all other methods you have an empty line after docstring. Add one here as well :)

Reply:
MRs welcome 😄